src/update_engine/utils.cc

```diff
@@ -402,6 +402,20 @@ bool StringHasPrefix(const std::string& str, const std::string& prefix) {
 const std::string BootDevice() {
   char boot_path[PATH_MAX];
   struct stat stbuf;
+  dev_t root_dev;
+
+  root_dev = rootdev_devt_from_mountpoint("/usr");
+  /* First search by mount point which works for btrfs and normal file systems,
+     then fall back to old behavior for propper error reporting */
+  if (root_dev == 0) {
+    /* Yields the containing dev_t in st_dev. */
+    if (stat("/usr", &stbuf) != 0) {
+      PLOG(ERROR) << "stat /usr failed:";
+      return "";
+    }
+    root_dev = stbuf.st_dev;
+  }
+
 
   if (stat("/usr", &stbuf) != 0) {
```

Comment:
Correct me if i'm wrong: this second stat is now unnecessary.

Reply:
Yes^^